launch_ros/launch_ros/actions/lifecycle_node.py

```diff
@@ -42,8 +42,9 @@ class LifecycleNode(Node):
     def __init__(
         self,
         *,
-        name: Optional[Text] = None,
-        node_name: Optional[Text] = None,
+        name: Optional[SomeSubstitutionsType] = None,
+        namespace: SomeSubstitutionsType,
+        node_name: Optional[SomeSubstitutionsType] = None,
```

Comment:
Would expect SyntaxError here

```python
>>> def foo(a=2, b, c=3):
...    print(f"{a} {b} {c}")
... 
  File "<stdin>", line 1
SyntaxError: non-default argument follows default argument
```

Reply:
AFAIK it doesn't apply to only keyword arguments, only to positional arguments

Reply:
Ah, sure enough

```python3
>>> def foo(*, a=2, b, c=3):
...    print(f"{a} {b} {c}")
... 
>>> foo(a=1, b=2, c=3)
1 2 3
```